workspace/src/telemetry_mqtt/telemetry_mqtt/mqtt_telemetry_publisher.py

```diff
@@ -50,25 +66,89 @@ def __init__(self):
         )
         connected_future = self.mqtt_conn.connect()
         connected_future.result()
+        self.get_logger().info("Connected!")
+
+    def connect_using_discovery(self, cert_data):
+        tries = 0
+
+        tls_options = io.TlsContextOptions.create_client_with_mtls_from_path(
+            cert_data["certificatePath"],
+            cert_data["privateKeyPath"],
+        )
+        tls_options.override_default_trust_store_from_path(None, cert_data["rootCAPath"])
+        tls_context = io.ClientTlsContext(tls_options)
+
+        discovery_client = DiscoveryClient(
+            io.ClientBootstrap.get_or_create_static_default(),
+            io.SocketOptions(),
+            tls_context,
+            # TODO: Get region from config!
+            "us-west-2",
```

Comment:
Could we setup the conf file before we merge? and pull in the retry timeout value also into it?

Reply:
Yes - done in latest commit